packages/packages/documents/src/sync/__tests__/sync-store.test.ts

```diff
@@ -0,0 +1,259 @@
+					...mockDocument.editor,
+					isSaving: true,
+				},
+			},
+		] );
+
+		// Act.
+		dispatchV1ReadyEvent();
+
+		// Assert.
+		expect( getCurrentDocument( store ).isSaving ).toBe( true );
+	} );
+
+	it( 'should sync saving state of a document on save', () => {
+		// Arrange.
+		mockV1DocumentsManager( [
+			makeMockV1Document(),
+		] );
+
+		// Populate the documents state.
+		dispatchV1ReadyEvent();
+
+		// Assert - Default state.
+		expect( getCurrentDocument( store ).isSaving ).toBe( false );
+
+		// Act.
+		dispatchCommandBefore( 'document/save/save' );
+
+		// Assert - On save start.
+		expect( getCurrentDocument( store ).isSaving ).toBe( true );
+		expect( getCurrentDocument( store ).isSavingDraft ).toBe( false );
+
+		// Act.
+		dispatchCommandAfter( 'document/save/save' );
+
+		// Assert - On save end.
+		expect( getCurrentDocument( store ).isSaving ).toBe( false );
+		expect( getCurrentDocument( store ).isSavingDraft ).toBe( false );
+	} );
+
+	it( 'should sync draft saving state of a document on save', () => {
+		// Arrange.
+		mockV1DocumentsManager( [
+			makeMockV1Document(),
+		] );
+
+		// Populate the documents state.
+		dispatchV1ReadyEvent();
+
+		// Assert - Default state.
+		expect( getCurrentDocument( store ).isSavingDraft ).toBe( false );
+
+		// Act.
+		dispatchCommandBefore( 'document/save/save', {
+			status: 'autosave',
+		} );
+
+		// Assert - On save start.
+		expect( getCurrentDocument( store ).isSaving ).toBe( false );
+		expect( getCurrentDocument( store ).isSavingDraft ).toBe( true );
+
+		// Act.
+		dispatchCommandAfter( 'document/save/save', {
+			status: 'autosave',
+		} );
+
+		// Assert - On save end.
+		expect( getCurrentDocument( store ).isSaving ).toBe( false );
+		expect( getCurrentDocument( store ).isSavingDraft ).toBe( false );
+	} );
+
+	it( 'should sync dirty state of a document on V1 load', () => {
+		// Arrange.
+		const mockDocument = makeMockV1Document();
+
+		mockV1DocumentsManager( [ {
+			...mockDocument,
+			config: {
+				...mockDocument.config,
+				revisions: {
+					current_id: 2,
+				},
+			},
+		} ] );
+
+		// Act.
+		dispatchV1ReadyEvent();
+
+		// Assert.
+		expect( getCurrentDocument( store ).isDirty ).toBe( true );
+	} );
+
+	it( 'should sync dirty state of a document on document change', () => {
+		// Arrange.
+		const mockDocument = makeMockV1Document();
+
+		mockV1DocumentsManager( [
+			mockDocument,
+		] );
+
+		// Populate the documents state.
+		dispatchV1ReadyEvent();
+
+		// Mock a change.
+		mockV1DocumentsManager( [ {
+			...mockDocument,
+			editor: {
+				...mockDocument.editor,
+				isChanged: true,
+			},
+		} ] );
+
+		// Assert - Default state.
+		expect( getCurrentDocument( store ).isDirty ).toBe( false );
+
+		// Act.
+		dispatchCommandAfter( 'document/save/set-is-modified' );
+
+		// Assert - After change.
+		expect( getCurrentDocument( store ).isDirty ).toBe( true );
+	} );
+} );
+
+function getCurrentDocument( store: Store<SliceState<Slice>> ) {
```

Comment:
```suggestion
function getActiveDocument( store: Store<SliceState<Slice>> ) {
```

+ Maybe check against the selector and then it will actually be a useful check if something changes in the state but the selector stays the same.

Reply:
Done